fe/fe-core/src/main/java/org/apache/doris/nereids/pattern/PatternDescriptor.java

```diff
@@ -63,6 +66,22 @@ public <OUTPUT_TYPE extends Plan> PatternMatcher<INPUT_TYPE, OUTPUT_TYPE> thenAp
         return new PatternMatcher<>(pattern, defaultPromise, matchedAction);
     }
 
+    /**
+     * Same as thenApply, but catch all exception and return null
+     */
+    public <OUTPUT_TYPE extends Plan> PatternMatcher<INPUT_TYPE, OUTPUT_TYPE> thenApplyNoThrow(
+            MatchedAction<INPUT_TYPE, OUTPUT_TYPE> matchedAction) {
+        MatchedAction<INPUT_TYPE, OUTPUT_TYPE> adaptMatchedAction = ctx -> {
+            try {
+                return matchedAction.apply(ctx);
+            } catch (Exception ex) {
+                LOG.warn(ex.getMessage());
```

Comment:
add some msg header for easy search and log exception stack for easy debug
```suggestion
                LOG.warn("nereids apply rule failed, because {}", ex.getMessage(), ex);
```